app/dojo.py

```diff
@@ -248,3 +251,113 @@ def print_unallocated(self,filename=None):
 			txt_file.write(output)
 			txt_file.close()
 			return ("\nData has been successfully saved to {}.txt\n" .format(filename))
+
+	def reallocate_person(self, emailaddress, new_roomname):
+		"""This method reallocates a person using their unique identifier,
+		in this case their email address, to a the specified new room
+
+		"""
+		new_room = None
+		person_reallocating = None
+		
+		#check if a person with the email address exists
+		for member in itertools.chain(self.all_staff,self.all_fellows):
+			if member.email == emailaddress:
+				person_reallocating = member
+
+		if person_reallocating is None:
+			return (text_format.CRED + "\nCould not find person with email {}!\n"
+				.format(emailaddress)
+				+ text_format.CEND)
+
+		#checks if the room exists
+		for room in itertools.chain(self.all_offices, self.all_livingspace):
+			if room.name == new_roomname:
+				new_room = room	
+
+		if new_room is None:
+			return (text_format.CRED + "\nThe room {} does not exist!\n"
+				.format(new_roomname)
+				+ text_format.CEND)
+
+		#reallocating to a livingspace
+		if new_room.room_type == "LIVING SPACE":
+			if person_reallocating.role == "STAFF":
```

Comment:
Flat is better than nested https://www.python.org/dev/peps/pep-0020/

`if person_reallocating == "STAFF" and new_room.room_type == "LIVING SPACE":
`do this and that`